receiver/kubeletstatsreceiver/kubelet/volume.go

```diff
@@ -84,3 +85,22 @@ func volumeInodesUsedMetric(prefix string, s stats.VolumeStats) *metricspb.Metri
 		s.InodesUsed,
 	)
 }
+
+func getLabelsFromVolume(volume v1.Volume) map[string]string {
+	switch {
+	// TODO: Support more types
+	case volume.ConfigMap != nil:
+		return map[string]string{labelVolumeType: "configMap"}
```

Comment:
Why do we need to allocate map here if we need just one value?

Reply:
For certain volume types, we should also look into syncing metadata other than the volume type. Right now the receiver syncs only type but I'd like to pull in other labels as well where it's useful.

Reply:
For example: https://github.com/signalfx/signalfx-agent/blob/master/pkg/monitors/kubernetes/volumes/pvdim.go#L129

Reply:
I'm just worried about unnecessary map allocations which are pretty expensive. I see it's being used in an iteration only after that. Can we pass existing map to this function or return a slice instead?

Reply:
Yes, I believe it should be possible to pass in an existing map. Will update.

Reply:
Updated to pass in labels map from `setExtraLabels` method.